platforms/jvm/jacoco/src/main/java/org/gradle/testing/jacoco/tasks/JacocoReportBase.java

```diff
@@ -202,6 +203,7 @@ public void executionData(TaskCollection tasks) {
      * @return class dirs to report coverage of
      */
     @Internal
+    @ToBeKeptEagerProperty(because = "Not settable FileCollection")
```

Comment:
❓ Can we deprecate this instead? It is only used, and should only be useful, in the task actions, so this could be a `private` or `protected` method instead. Or just be inlined.

We can do this in two parts, and simply mark the method to be deprecated via an additional parameter to the annotation, like so:

```java
@ToBeKeptEagerProperty(because = "Not settable FileCollection", willBeDeprecated = true)
```

Reply:
Added willBeDeprecated = true